pkg/server/container_stop.go

```diff
@@ -65,10 +65,19 @@ func (c *criService) stopContainer(ctx context.Context, container containerstore
 	task, err := container.Container.Task(ctx, nil)
 	if err != nil {
 		if !errdefs.IsNotFound(err) {
+			// For the container running case we've somehow ended up in a state where the shim is gone but cri
+			// still thinks the container is running. If the task exit event was missed from events.handleContainerExit this
+			// is the likely culprit but a repro of this is hard to pull off. Even in the event of a shim crash task.Wait would return and
+			// update the status, so if this condition is hit it's a myriad of strange behavior probably
+			// related to containerd restart quirks. Simply reuse the logic if the state is unknown to update the status
+			// to so remove goes smoothly.
+			if state == runtime.ContainerState_CONTAINER_RUNNING {
+				return cleanupUnknownContainer(ctx, id, container)
+			}
 			return errors.Wrapf(err, "failed to get task for container %q", id)
 		}
 		// Don't return for unknown state, some cleanup needs to be done.
-		if state == runtime.ContainerState_CONTAINER_UNKNOWN {
+		if state == runtime.ContainerState_CONTAINER_UNKNOWN || state == runtime.ContainerState_CONTAINER_RUNNING {
 			return cleanupUnknownContainer(ctx, id, container)
```

Comment:
We should log here that CRI thinks the container is still running, but there's no task.

Reply:
Good idea